optimizer_wrapper.rb

```diff
@@ -198,10 +198,12 @@ def self.solve(service_vrp, job = nil, block = nil)
 
     if !vrp.subtours.empty?
       multi_modal = Interpreters::MultiModal.new(vrp, service)
-      optim_result = multi_modal.multimodal_routes
-    elsif vrp.vehicles.empty? || (vrp.services.empty? && vrp.shipments.empty?)
-      optim_result = config[:services][service].empty_result(
+      multi_modal_result = multi_modal.multimodal_routes
+      optim_result = parse_result(vrp, multi_modal_result)
+    elsif vrp.vehicles.empty? || (vrp.services.empty? && vrp.shipments.empty?) # parse result here too
+      empty_result = config[:services][service].empty_result(
         service.to_s, vrp, 'No vehicle available for this service', false)
+      optim_result = parse_result(vrp, empty_result)
```

Comment:
Can't we make sure that `multimodal_routes` and `empty_result` call `parse_result` so that they generate conforming results and we don't need to "remember calling parse_result each time we call these result functions"